### ISSUE TITLE
Code duplication in vanilla policy gradients

### ISSUE THREAD

**Issue by HenryJia** [PyTorchLightning/pytorch-lightning-bolts#182]
## 🐛 Bug

It seems like there's 2 near identical implementations of VPG
https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/master/pl_bolts/models/rl/reinforce_model.py
https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/master/pl_bolts/models/rl/vanilla_policy_gradient_model.py

**Comment by Borda** [comment]
@HenryJia mind send a PR? :]

**Comment by sid-sundrani** [comment]
That is true. Also, the implementation of VPG in vanilla_policy_gradient_model.py collects and trains the agent on a trajectory only from one episode each time, whereas reinforce trains on trajectories from multiple episodes

**Comment by Borda** [comment]
@sid-sundrani @HenryJia anyone wants to take it over and send a PR?

**Comment by sid-sundrani** [comment]
Sure. 

Are we looking to remove one of these implementations? If so I think `Reinforce` could be better to keep- simpler loss, trains on multiple trajectories, and would make a good baseline. Also, `VanillaPolicyGradient ` on Cartpole with default setting doesn't seem to be learning (`reward`=~10, `avg_reward`=~9.43 is constant since the start)
